src/EditorFeatures/CSharpTest2/Recommendations/ElseKeywordRecommenderTests.cs

```diff
@@ -86,43 +90,218 @@ public class ElseKeywordRecommenderTests : KeywordRecommenderTests
         }
 
         [Fact, Trait(Traits.Feature, Traits.Features.KeywordRecommending)]
-        public async Task TestAfterIf()
+        public async Task TestNotAfterIf()
         {
-            await VerifyKeywordAsync(AddInsideMethod(
+            await VerifyAbsenceAsync(AddInsideMethod(
 @"if (true)
-    Console.WriteLine();
 $$"));
         }
 
-        [Fact, Trait(Traits.Feature, Traits.Features.KeywordRecommending)]
-        public async Task TestAfterIfBlock()
+        class Statements : DataAttribute
+        {
+            public override IEnumerable<object[]> GetData(MethodInfo testMethod)
+                => new[] { new[] { "Console.WriteLine();" }, new[] { "{ }" }, new[] { "while (true) { }" } };
+        }
+
+        [Theory, Trait(Traits.Feature, Traits.Features.KeywordRecommending)]
+        [Statements]
+        public async Task TestAfterIfStatement(string statement)
         {
             await VerifyKeywordAsync(AddInsideMethod(
-@"if (true)
-{
-    Console.WriteLine();
-}
+$@"if (true)
+    {statement}
```

Comment:
i personally dislike this.  I cannot see what's happening.  My tests are now non-local.  

The only sort of 'theory' approach i've ever been ok with is targetted, simple application of theories, where the theory is on hte test method itself.  if i have to look up more than 2-3 lines, or i have to go look at another *class*/*method* etc. then i've completely lost the plot on what the test is testing.  Previously i could see this exactly.  Now i can't.  this is a net negative for me here.

Reply:
ok I can revert it back... it just seemed to me like this way it's much easier to read, review and spot bugs in the tests, whereas with a lot of duplication it's really easy to miss something

Reply:
and also more importantly, when writing more tests people will usually avoid writing all the cases and just do some of them to not bother with writing so much duplication, which can sometimes miss bugs due to missing test cases

Reply:
IMO, tests for this sort of thing should be written based on the expected interesting cases based on the code change.  So, your change adds an arbitrary walk upwards looking for if-statements.  And it also checks that if-statement for the last token of the true-statement within it.

It seems like we could test this out with some very judicious tests that would demonstrate that both of those pieces of logic are working properly.  For example, we sholud not see 'else' here:

```c#
if (true) 
     Console.WriteLine()
$$
```

Because the last token of hte true-statement would be the missing-token ```;``` and htat's not what we're on.  Similarly, we should see if after ```Console.WriteLine();```.  And we should see if htis if it deeply nested.   Testing one level deep, two levels deep, and something larger should be sufficient.

This does not seem like a lot of tests needed.  It tests positive and negative cases.  And feels appropriate given the code change.

Reply:
Thanks, I'll add tests for that too.

Reply:
> it just seemed to me like this way it's much easier to read, review and spot bugs in the tests

It's harder for me, because i ca'nt just look at a test and know what it means.  I have to go analyze other parts of the test file to know.  We try to avoid that.  I've accepted some Theories in hte past where the theory *travels* with the test method.  and the theory itself is very tiny.  i.e. i've seen theories that try to do major replacements in the code, and def make it less clear. 

I'm on the fence here about this one.  But, give my previous post, i don't think it adds substantial value given just writing the 6ish test cases manually.

Reply:
Yes.  this form seems ok to me.  i can live with inline data at individual test sites.